src/tests/test_name_constraint.cpp

```diff
@@ -0,0 +1,96 @@
+/*
+* (C) 2015,2016 Kai Michaelis
+*
+* Botan is released under the Simplified BSD License (see license.txt)
+*/
+
+#include "tests.h"
+
+#if defined(BOTAN_HAS_X509_CERTIFICATES)
+  #include <botan/x509path.h>
+  #include <botan/internal/filesystem.h>
+#endif
+
+#include <algorithm>
+#include <fstream>
+#include <iomanip>
+#include <string>
+#include <vector>
+#include <map>
+#include <cstdlib>
+
+namespace Botan_Tests {
+
+namespace {
+
+#if defined(BOTAN_HAS_X509_CERTIFICATES)
+
+class Name_Constraint_Tests : public Test
+   {
+   public:
+      std::vector<Test::Result> run() override
+         {
+         const std::vector<std::tuple<std::string,std::string,std::string,std::string>> test_cases = {
```

Comment:
The root is valid for 10 years but the end certs only have a 1 year lifespan. To avoid having to regen these certs again in the near future can you give the end certs also a ~10 year period, and use SHA-256 instead of SHA-1 so these tests don't break when support for SHA-1 signatures is disabled.